### ISSUE TITLE
mypy does not error when calling a `TypeAliasType`, if it refers to `dict`

### ISSUE THREAD

**Issue by sbrudenell** [python/mypy#19698]
**Bug Report**

If I declare a type with `type T = dict`, then instantiate it with `T()`, mypy does not error.

I spot-checked some other types like `type T = set`, `type T = int` etc, and mypy complains about these when called. Only `type T = dict` seems to wrongly pass.

**To Reproduce**

```python
type T = dict
x: T = T()
```

**Expected Behavior**

```shell
$ mypy t.py
t.py:2: error: "TypeAliasType" not callable  [operator]
Found 1 error in 1 file (checked 1 source file)
```

**Actual Behavior**

```shell
$ mypy t.py
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: `mypy 1.17.1 (compiled: yes)`
- Mypy command-line flags: `mypy`
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: `Python 3.13.5`

**Comment by brianschubert** [comment]
Thanks for the report! I think I found the cause. Here's another funny effect this has:
```python
type MyType = type

def foo(x: object) -> None:
    if MyType(x) is str:
        reveal_type(x)  # N: Revealed type is "builtins.str"
```
